.github/workflows/go.yaml

```diff
@@ -52,5 +68,13 @@ jobs:
       - name: Build sdk-features go
         run: go build
 
+      - name: Start server & deps
+        if: ${{ inputs.docker-image-artifact-name }}
+        run: docker-compose -f /tmp/server-docker/docker-compose.yml -f ./dockerfiles/docker-compose.for-server-image.yaml up -d temporal-server cassandra elasticsearch
+
       - name: Run SDK-features tests
-        run: go run . run --lang go --version "${{ inputs.version-is-repo-ref && '$(realpath ../sdk-go)' || inputs.version }}"
+        run: go run . run --lang go ${{ inputs.docker-image-artifact-name && '--server localhost:7233 --namespace default' || ''}} --version "${{ inputs.version-is-repo-ref && '$(realpath ../sdk-go)' || inputs.version }}"
+
+      - name: Tear down docker compose
+        if: inputs.docker-image-artifact-name && (success() || failure())
+        run: docker-compose -f /tmp/server-docker/docker-compose.yml -f ./dockerfiles/docker-compose.for-server-image.yaml down -v
```

Comment:
My opinion about CI workflows is that they should be reproducible locally (for development and debugging) and should be documented for contributors.

It's hard to maintain (keep them in sync) three separate parts: documentation, CI workflow steps, and local development workflow. I suggest having a single command via build manager `make`, `mage`, or similar. And everywhere have a simple command such as `make start-server-and-deps; make test; make teardown`. And then have a single place for maintaining them.

Reply:
I don't want to introduce `make` since we haven't needed it yet. That said I'd be fine with adding a few little bash scripts into the `.github` directory that could be used for this purpose.

The commands cannot be combined into one line without breaking the `if` guards, which are important for visibility purposes in the CI UI. I wouldn't want someone to need to walk the script just to understand why something is or is not happening.

That sound good?

Reply:
I would strongly discourage `make` as it is not a good cross platform tool. We have a CLI right here called sdk-features. If we have to move this simple one line to inside of the CLI, ok. You're not gaining much though.

Reply:
Given these bits are only intended to run on CI, I'd just take the ease of a shell script here

Reply:
> Given these bits are only intended to run on CI

Conflicts with

> My opinion about CI workflows is that they should be reproducible locally

If they're only for CI, they can be in the CI language of choice (YAML). If the requirement is "reproducible locally" we should use this repo's language of choice (Go) and not something platform specific.

Reply:
Well, because this
> Why is docker-compose + args not ok but make + args ok?

Was basically my thought. Though we can change it to "deps up" and "deps down" or something instead of passing the docker files & args. I agree it's not much savings. I mean, I can just move it into Go and be done with it but it does feel a little overkill. @mindaugasrukas are you fine with that?

Reply:
`make` was mentioned here as an example. Bash or other scripts is fine.

My point here is it's hard to keep documentation in sync with CI workflow and development workflow definitions. Also, having CI workflow execution steps defined in a YAML makes it complicated to lint, test, and reproduce them locally. Yes, it's only a few lines of code today, but there will be many more tomorrow. And then reproduction locally becomes many copy-pasting with potential errors.

An alternative (which works for me) is to call a script inside workflows definitions: I.e., `run: bash setup-something.sh <params...>`, `run: bash run-test.sh <params>` and similar, and move all the logic to those scripts. That makes it easy to keep all workflows in sync, lint command, test logic, and reproduce CI runs locally. All parameters might be defined as env variables. So documentation and reproduction steps locally might look like this:
```
export PARAM_1=value-1
export PARAM_2=value-1
export PARAM_3=value-1
...
bash setup-something.sh
bash run-test.sh
bash cleanup.sh
...
```

This is a proposal from my personal experience. If that something is not applicable here, it's fine to keep it as is.

Reply:
Another argument for scripts is that there are four duplicated commands in this PR, and they all must be in sync. Having scripts makes them easy to maintain.

Reply:
Yeah, I absolutely agree with the philosophy here, I just think we're not quite at that point with the CI definitions. They're all one liners, and the way you'd run the docker compose stuff specifically locally is necessarily different from in CI since in CI the images are downloaded from the last  run's artifacts, which you can't do locally. If they were more than one-line invocations then yeah I think we'll want to move them inside the Go runner.

Then the invocation of the actual test is just a normal one-liner you'd want to control yourself locally.

As for the repetition across the definitions per language, yeah, that's a bit annoying. If/when these get more complex, I think solving that by templating them is probably the most straightforward way.